starter-cli/src/main/java/io/micronaut/starter/cli/feature/picocli/CreateCommandCommand.java

```diff
@@ -95,13 +103,48 @@ public Integer call() throws Exception {
             }
         }
 
-        if (config.getTestFramework() == TestFramework.JUNIT) {
-            renderResult = templateRenderer.render(junit.getTemplate(config.getSourceLanguage(), project), overwrite);
-        } else if (config.getTestFramework() == TestFramework.SPOCK) {
-            renderResult = templateRenderer.render(spock.getTemplate(project), overwrite);
-        } else if (config.getTestFramework() == TestFramework.KOTLINTEST) {
-            renderResult = templateRenderer.render(kotlinTest.getTemplate(project), overwrite);
+        List<Language> supportedLanguagesForTestFramework = new ArrayList<>();
+        Language defaultLanguageForTestFramework = null;
+        switch (config.getTestFramework()) {
+            case KOTLINTEST:
+                defaultLanguageForTestFramework = Language.KOTLIN;
+                supportedLanguagesForTestFramework.add(defaultLanguageForTestFramework);
+                break;
+
+            case SPOCK:
+                defaultLanguageForTestFramework = Language.GROOVY;
+                supportedLanguagesForTestFramework.add(defaultLanguageForTestFramework);
+                break;
+
+            case JUNIT:
+                defaultLanguageForTestFramework = Language.JAVA;
+                supportedLanguagesForTestFramework.add(defaultLanguageForTestFramework);
+                supportedLanguagesForTestFramework.add(Language.KOTLIN);
+                supportedLanguagesForTestFramework.add(Language.GROOVY);
+                break;
+
+            default:
+                defaultLanguageForTestFramework = Language.JAVA;
+                supportedLanguagesForTestFramework.add(defaultLanguageForTestFramework);
+
         }
+        Language rockerTemplateLanguage = (supportedLanguagesForTestFramework.contains(config.getSourceLanguage())) ?
```

Comment:
If this gets merged: https://github.com/micronaut-projects/micronaut-starter/pull/415

this code could be rewritten as: 

```Language rockerTemplateLanguage = (config.getTestFramework().getSupportedLanguages().contains(config.getSourceLanguage())) ?
            config.getSourceLanguage() : config.getTestFramework().getDefaultLanguage();```

and the above block could be deleted.